agent/tcs/client/client_test.go

```diff
@@ -134,8 +138,17 @@ func TestPayloadHandlerCalled(t *testing.T) {
 		}
 	}()
 
-	time.Sleep(1 * time.Millisecond)
-	if handledPayload == nil {
+	go func() {
+		time.Sleep(20 * time.Millisecond)
+		timeout <- true
+	}()
+
+	select {
+	case <-requestComplete:
+		if handledPayload == nil {
```

Comment:
I agree that we don't need both `requestComplete` and `handledPayload` here. It would be better to just make handledPayload a channel and pass back the payload received to that. In the `select` loop, you can just wait until you get a message on that channel.